src/app/group/group-add-edit/group-add-edit.component.ts

```diff
@@ -11,68 +14,149 @@ import { filter } from 'rxjs';
         this.isGroupAddActive = true;
       }
-      else{
+      else {
         this.isGroupAddActive = false;
       }
     })
-  }
+
+
+  }// OnInit method end
 
   // Add member name and member email formgroup in add group form
-  memberNameAndEmailForm(): FormGroup{
+  addMembersToGroupForm(): FormGroup {
     return new FormGroup({
-      memberName : new FormControl(''),
-      memberEmail : new FormControl('')
+      memberName: new FormControl(''),
+      memberEmail: new FormControl('')
     })
   }
 
   // Getter methods
-  get groupName(){
-    return this.addGroupForm.get('groupName');
+  /** Get groupName FormControl
+   *  @returns FormControl
+   **/
+  get groupName() : FormControl {
+    return this.addGroupForm.get('groupName') as FormControl;
+  }
+
+  /** Get groupDescription FormControl
+   *  @returns FormControl
+   **/
+  get about() : FormControl {
+    return this.addGroupForm.get('about') as FormControl;
+  }
+
+  /** Get groupMember FromArray
+   *  @returns FormArray
+   **/
+  get groupMembers() : FormArray {
+    return this.addMembersForm.get('groupMembers') as FormArray;
   }
 
-  // getter of dyanamic create group member
-  get memberNameEmail(){
-    return this.addGroupForm.get('groupMember') as FormArray;
+  /** Add member field in form
+   *  on click of add menber button
+   **/
+  addMemberNameAndEmailField(): void {
+    this.groupMembers.push(this.addMembersToGroupForm());
   }
 
-  // Add member dynamically
-  addMemberNameAndEmail(): void{
-    this.memberNameEmail.push(this.memberNameAndEmailForm());
+  /** Delete member field in form
+   *  on click of Delete button
+   *  @param index
+   **/
+  deleteMemberNameAndEmailField(i: number): void {
+    this.groupMembers.removeAt(i);
   }
 
-  // Delete member name and email
-  deleteMemberNameAndEmail(i : number): void{
-    this.memberNameEmail.removeAt(i);
+  /** create group method to get data from add group form
+   * and send to post api call function
+   **/
+  createGroup() {
+    const data : Groups = {
+      name: this.groupName?.value as string,
+      about: this.about?.value as string,
+      groupMembers: [{
+          userId: "3a0ba79a-0f40-0dd1-4e91-9ed501777180"
+        }]
+    }
+
+    // Create group Api call from group service
+    this.groupsService.createGroup(data).subscribe(() => {
+      alert('Group created successfully!');
+    });
+    this.isGroupCreated = true;
+  }
+
+  /** add Members method
+   * to get data from add add members form
+   * and send to post api call function
+   **/
+  addMembersToGroup() {
+    const data : GroupMembersToAdd = this.groupMembers.value;
+
+    // AddMembers api call from group service
+    this.groupsService.addMembersToGroup(data).subscribe(() => {
+      alert('members added successfully!');
+    });
+  }
+
+
+  /** current user method
+   * to get name of current user
+   * from server through api
+   * @returns string
+   * */
+  getCurrentUser() : string {
```

Comment:
What will be return in the method?
Example: @returns string Returns the user name